cmd/docker/docker.go

```diff
@@ -37,7 +37,8 @@ func main() {
 	dockerCli, err := command.NewDockerCli(command.WithBaseContext(ctx))
 	if err != nil {
 		fmt.Fprintln(os.Stderr, err)
-		os.Exit(1)
+		defer os.Exit(1)
```

Comment:
Why defer if we're doing `return` immediately after?

Reply:
Oh, I actually had the same question and then forgot to post the comment 😓

Reply:
@Benehiko

Reply:
because we `defer cancel()`. The linter warned that the defer might not run if we only do `os.Exit()` so I moved them into a defer call and then rather return the main function.

Reply:
the other option is to move everything with the signal handler into a function which returns an error, which then does `os.Exit(statusCode)` inside `main()`.

```go
func main() {
 err := doSomething()
 if err != nil {
   os.Exit(1)
 }
}

func doSomething() error {

  ctx, cancel := ...
  defer cancel() 
  
  ...
}

Reply:
IMO, having an additional "C style" main would be nice:

```go
func main() {
    ret := dockerMain()
    if ret != 0 {
        os.Exit(ret)
    }
}

func dockerMain() int {
     ... all the stuff from the current main
     ... but return <exit code> instead of calling os.Exit
}

```

Reply:
Or just:

```go
func main() {
  var exitCode int
  defer func() {
    if exitCode != 0 {
      os.Exit(exitCode)
    }
  }()
  ...
}

Reply:
`defer os.Exit()` will only silence the linter, but since defers are executed (bottom to top), it would still do the exit before all the previous defers.

Reply:
hmm true, alright i'll refactor it